src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs

```diff
@@ -19,12 +19,14 @@ public static class TextLoaderSaverCatalog
         /// <param name="hasHeader">Whether the file has a header.</param>
         /// <param name="separatorChar">The character used as separator between data points in a row. By default the tab character is used as separator.</param>
         /// <param name="dataSample">The optional location of a data sample. The sample can be used to infer column names and number of slots in each column.</param>
+        /// <param name="allowSparse">Whether the file can contain numerical vectors in sparse format.</param>
         public static TextLoader CreateTextLoader(this DataOperationsCatalog catalog,
             TextLoader.Column[] columns,
             bool hasHeader = TextLoader.DefaultArguments.HasHeader,
             char separatorChar = TextLoader.DefaultArguments.Separator,
-            IMultiStreamSource dataSample = null)
-            => new TextLoader(CatalogUtils.GetEnvironment(catalog), columns, hasHeader, separatorChar, dataSample);
+            IMultiStreamSource dataSample = null,
+            bool allowSparse = TextLoader.DefaultArguments.AllowSparse)
```

Comment:
>TextLoader [](start = 31, length = 10)

Consider that since this is the code that needs to be changed for #2452, which you've self-assigned anyway, we may as well correct this while we're at it. (Otherwise I see no point in changing it.)

Reply:
Couldn't agree more. Proposed solution to #2452 can be found in Iteration 4!

---
In reply to: [258240786](https://github.com/dotnet/machinelearning/pull/2630#discussion_r258240786) [](ancestors = 258240786)